src/react-components/mic-level-widget.js

```diff
@@ -3,80 +3,55 @@ import PropTypes from "prop-types";
 import { FormattedMessage } from "react-intl";
 import MovingAverage from "moving-average";
 
-const AudioContext = window.AudioContext || window.webkitAudioContext;
-
 export default class MicLevelWidget extends Component {
   static propTypes = {
-    mediaStream: PropTypes.object,
     hasAudioTrack: PropTypes.bool,
-    muteOnEntry: PropTypes.bool
+    muteOnEntry: PropTypes.bool,
+    scene: PropTypes.object
   };
 
   state = {
-    micLevel: 0
+    volume: 0
   };
 
   componentDidMount() {
-    if (this.props.mediaStream != null && this.props.hasAudioTrack) {
-      this.startAnalyzer(this.props.mediaStream);
-    }
-  }
-
-  componentDidUpdate(prevProps) {
-    if (this.props.mediaStream !== prevProps.mediaStream || this.props.hasAudioTrack !== prevProps.hasAudioTrack) {
-      this.stopAnalyzer();
-      if (this.props.mediaStream != null && this.props.hasAudioTrack) {
-        this.startAnalyzer(this.props.mediaStream);
-      }
-    }
+    this.startAnalyser();
   }
 
   componentWillUnmount() {
-    this.stopAnalyzer();
+    this.stopAnalyser();
   }
 
-  stopAnalyzer() {
+  stopAnalyser() {
     if (this.micUpdateInterval != null) {
       clearInterval(this.micUpdateInterval);
     }
-    if (this.micLevelAudioContext != null) {
-      this.micLevelAudioContext.close();
-    }
   }
 
-  startAnalyzer(mediaStream) {
-    this.micLevelAudioContext = new AudioContext();
-    const micSource = this.micLevelAudioContext.createMediaStreamSource(mediaStream);
-    const analyser = this.micLevelAudioContext.createAnalyser();
-    analyser.fftSize = 32;
-    const levels = new Uint8Array(analyser.frequencyBinCount);
-    micSource.connect(analyser);
+  startAnalyser() {
+    if (this.micUpdateInterval) {
+      clearInterval(this.micUpdateInterval);
+    }
     const micLevelMovingAverage = MovingAverage(100);
+    let max = 0;
     this.micUpdateInterval = setInterval(() => {
-      analyser.getByteTimeDomainData(levels);
-      let v = 0;
-      for (let x = 0; x < levels.length; x++) {
-        v = Math.max(levels[x] - 128, v);
-      }
-      const level = v / 128.0;
-      // Multiplier to increase visual indicator.
```

Comment:
I made it so that the volume indicators glow in proportion to the loudest sound you've made in the microphone so far. This max should probably be a "max in the last minute or so" so that if you make a very loud noise that's well above your speaking voice, then your glow bars will recover after a minute. It's fine as-is for now and fills up well for non-degenerate cases like described above.

Reply:
Changed micLevelsForVolume such that the icon sees a wide range when talking.